### PR TITLE
Add 3.13 support, remove 3.8

### PR DESCRIPTION
**Pull request by ChrisLovering**
CI will fail on 3.13 builds until https://github.com/actions/python-versions/pull/312 is merged to add 3.13 support to setup-python

### CODE REVIEW

File: .github/workflows/run-cibuildwheel.yml [hunk 1]
```diff
@@ -55,7 +55,7 @@ jobs:
         run: echo "CIBW_ARCHS=${{ matrix.cibw_archs }}" >> $GITHUB_ENV
         shell: bash
       - name: Build wheels
-        uses: pypa/cibuildwheel@v2.16.2
+        uses: pypa/cibuildwheel@v2.21.2
```
Comment (ChrisLovering) [inline .github/workflows/run-cibuildwheel.yml:58]: This version is both the latest as of writing, and the one that adds 3.13 support
https://github.com/pypa/cibuildwheel/releases

**Comment by ChrisLovering** [comment]
https://github.com/actions/python-versions/pull/312 has been merged, and 3.13 CI has passed here. Unsure why the macOS runners aren't picking up the outstanding jobs though.

@saghul this is ready for revuew :)

**Comment by ChrisLovering** [comment]
macOS CI failed as no runner picked up the jobs within a day, rather than due to an error in the change itself.

That doesn't mean there isn't a problem with these changes, just that we don't know yet.

**Comment by saghul** [comment]
macOS 11 was decomissioned, so we need to update https://github.com/saghul/pycares/blob/57a82a0039ad2607092abe6aff0f06be17a400f2/.github/workflows/test.yml#L12

**Comment by ChrisLovering** [comment]
Updated CI to run using macOS 13 (newest x86) and macOS 15 for M1 support.

Various mac versions are failing the `test_custom_resolvconf` test due to the errorno not being the expected value. Unsure on the cause of this, as some version of Python finish tests just fine, and the ones that fail aren't consistent across macOS 13 & 15.

**Comment by candrews** [comment]
Can we get this merged and released?

I'm really looking forward to Python 3.13 support :)

Thank you!

**Comment by saghul** [comment]
Tests are failing, any help with that would be welcome!

**Comment by ChrisLovering** [comment]
Yeah, it looks like it's a potential flaky test, as it works on some Python versions on macOS 13 and then other Python versions on macOS 15.

I don't have a mac so can't reproduce in order to fix it.

**Comment by Zaczero** [comment]
It doesn't look like the failing tests are related to the Python 3.13 upgrade.

**Comment by No767** [comment]
I've been looking into the tests and my best guess is that it has to do with something with MacOS or how the tests are set up

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -9,8 +9,8 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-20.04, windows-2019, macos-11]
-        python-version: ['3.8', '3.9', '3.10', '3.11', '3.12']
+        os: [ubuntu-20.04, windows-2019, macos-13, macos-15]
```
Comment (No767) [inline .github/workflows/test.yml:12]: Something of note is that we should be testing against Windows 2022 and Ubuntu 24.04 if we are upgrading macos versions as well

**Comment by monosans** [comment]
Could you please upload CPython 3.13 wheels to PyPI?

**Comment by ChrisLovering** [comment]
hey @saghul is there anything blocking making a release to pypi for this PR that we can help with?

**Comment by saghul** [comment]
I tried to publish them but ran into a problem. Looks like a v3 / v4 upload action shenanigans.

I'll try to look into it soon.

**Pull request merged**